packages/node/src/methods/install-operations.ts

```diff
@@ -82,8 +104,175 @@ export async function addAppInstance(
   params.peerAddress = nodeMsg.from!;
   delete params.proposal;
   if (nodeMsg.data.proposal) {
-    await this.channels.setAppInstanceIDForProposeInstall(params);
+    await setAppInstanceIDForProposeInstall(
+      this.selfAddress,
+      this.store,
+      params
+    );
   } else {
-    await this.channels.install(params);
+    await install(this.store, params);
+  }
+}
+
+/**
+ * Creates a ProposedAppInstanceInfo to reflect the proposal received from
+ * the client.
+ * @param selfAddress
+ * @param store
+ * @param params
+ */
+async function createProposedAppInstance(
+  selfAddress: Address,
+  store: Store,
+  params: Node.ProposeInstallParams
+): Promise<string> {
+  const appInstanceId = generateUUID();
+  const channel = await getChannelFromPeerAddress(
+    selfAddress,
+    params.peerAddress,
+    store
+  );
+
+  const proposedAppInstance = new ProposedAppInstanceInfo(
+    appInstanceId,
+    params
+  );
+
+  await store.addAppInstanceProposal(channel, proposedAppInstance);
+  return appInstanceId;
+}
+
+async function install(
+  store: Store,
+  params: Node.InstallParams
+): Promise<AppInstanceInfo> {
+  if (!params.appInstanceId) {
+    return Promise.reject("No AppInstanceId specified to install");
+  }
+
+  const stateChannel = await store.getChannelFromAppInstanceID(
+    params.appInstanceId
+  );
```

Comment:
Stylistically, you can do `const { appInstanceId } = params;` above and then this and the following line won't need to be 3 lines long